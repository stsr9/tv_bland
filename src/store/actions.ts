import { Schedule } from '../models/schedule'
import ScheduleAPI from '../api/schedule-api'
import ShowAPI from '../api/shows-api'
import { Show } from '../models/show'
import { Actor } from '../models/actor'

export enum ActionTypes {
    fetchEpisodes = 'FETCH_EPISODES',
    fetchEpisodeDetails = 'FETCH_EPISODE_DETAILS',
	fetchCast = 'FETCH_CAST',
	clearDetails = 'CLEAR_DETAILS'
}

interface EpisodeActions {
	fetchEpisodes: Function
	fetchEpisodeDetails: Function
	clearDetails: Function
}

interface CastActions {
	fetchCast: Function
}

export const episodeActions: EpisodeActions = {
	fetchEpisodes: () => async(dispatch: Function) => {
		const episodes: Schedule[] = await ScheduleAPI.fetchAllEpisodes()
	    dispatch({
			type: ActionTypes.fetchEpisodes,
			episodes
	    })
	},
	fetchEpisodeDetails: (id: string) => async(dispatch: Function) => {
		const episodeDetails: Show = await ShowAPI.fetchEpisodeDetails(id)
	    dispatch({
			type: ActionTypes.fetchEpisodeDetails,
			episodeDetails
	    })
	},
	clearDetails: () => (dispatch: Function) => {
		dispatch({
			type: ActionTypes.clearDetails,
	    })
	}
}

export const castActions: CastActions = {
	fetchCast: (id: string) => async(dispatch: Function) => {
		const cast: Actor[] = await ShowAPI.showCast(id)
	    dispatch({
			type: ActionTypes.fetchCast,
			cast
	    })
	}
}
