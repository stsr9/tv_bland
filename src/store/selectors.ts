import { MainState } from 'models/main-state'

export const episodeSelector = {
	getEpisodeList: ({ episodesState }: MainState) => episodesState.episodes,
	getEpisodeDetails: ({ episodesState }: MainState) =>
		episodesState.episodeDetails
}

export const castSelector = {
	getCast: ({ castState }: MainState) => castState.cast
}
