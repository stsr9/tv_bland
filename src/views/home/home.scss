.home {
  p {
    color: gray;
    font-size: 1.2em;
    width: 30vw;
    @media (max-width: 800px) {
      width: 90vw;
    }
  }
  .view-header {
    @media (max-width: 800px) {
      padding-bottom: 3vh;
    }
  }
  .episode-list {
    margin-top: -10vh;
    @media (max-width: 800px) {
      margin-top: 3vh;
    }
  }
  .episode-grid {
    padding-top: 3vh;
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(15%, 1fr));
    align-items: flex-start;
    grid-gap: 1rem;
    @media (max-width: 1600px) {
      grid-template-columns: repeat(5, minmax(15%, 1fr));
    }
    @media (max-width: 1400px) {
      grid-template-columns: repeat(4, minmax(15%, 1fr));
    }
    @media (max-width: 1200px) {
      grid-template-columns: repeat(3, minmax(15%, 1fr));
    }
    @media (max-width: 800px) {
      grid-template-columns: repeat(2, minmax(15%, 1fr));
      grid-gap: .7rem;
    }

  }
}
