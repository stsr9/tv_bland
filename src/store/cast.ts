import { AnyAction } from 'redux'
import { ActionTypes } from './actions'
import { Actor } from 'models/actor'

export interface InitialState {
	cast: Actor[]
}

const initialState: InitialState = {
	cast: []
}

export const castState = (
	state: InitialState = initialState,
	action: AnyAction
) => {
	switch (action.type) {
		case ActionTypes.fetchCast:
			return {
				...state,
				cast: action.cast
			}
		default:
			return state
	}
}
