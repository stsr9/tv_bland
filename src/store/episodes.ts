import { AnyAction } from 'redux';
import { Schedule } from '../models/schedule'
import { Show } from '../models/show';
import { ActionTypes } from './actions'

type InitialState = {
	episodes: Schedule[]
	episodeDetails: Show | null
}

const initialState: InitialState = {
	episodes: [],
	episodeDetails: null
};

export const episodesState =  (state: InitialState = initialState, action: AnyAction) => {
	switch (action.type) {
		case ActionTypes.fetchEpisodes:
			return {
				...state,
                episodes: action.episodes
			};
		case ActionTypes.fetchEpisodeDetails:
			return {
				...state,
                episodeDetails: action.episodeDetails
			};
		case ActionTypes.clearDetails:
			return {
				...state,
                episodeDetails: null
			};
		default:
			return state;
	}
};
