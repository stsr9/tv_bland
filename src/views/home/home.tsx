import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Schedule } from 'models/schedule'
import { episodeActions } from 'store/actions'
import { episodeSelector } from 'store/selectors'
import { SplashScreen } from 'components/splash-screen/splash-screen'
import { Episode } from 'components/episode/episode'
import { useTranslation } from 'react-i18next'
import './home.scss'

export const Home = () => {
    const episodes: Schedule[] = useSelector(episodeSelector.getEpisodeList)
    const dispatch = useDispatch()
    const { t } = useTranslation('common')

    useEffect(() => {
        if(!episodes.length) {
            dispatch(episodeActions.fetchEpisodes())
        }
	}, [dispatch, episodes]);

    return episodes.length
        ? (
            <div className="home">
                <header className="view-header">
                    <p>{t('homeHeadline')}</p>
                    <p>{t('homeHeaderText')}</p>
                </header>
                <div className="view-content">
                    <div className="episode-list">
                        <h2 className="fwn">{t('episodeListTitle')}</h2>
                        <div className="episode-grid flex">
                            {episodes.map((episode) =>
                                <Episode key={episode.id} episode={episode} />
                            )}
                        </div>
                    </div>
                </div>
            </div>
        )
        :  <SplashScreen />

}
