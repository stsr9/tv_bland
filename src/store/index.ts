import { combineReducers } from 'redux';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { episodesState } from './episodes'
import { castState } from './cast'

const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const appReducers = combineReducers({
	episodesState,
	castState
});

export const store = createStore(
	appReducers,
	composeEnhancers(applyMiddleware(thunk))
);

