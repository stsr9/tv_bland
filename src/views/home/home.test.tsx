import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { applyMiddleware, createStore} from 'redux';
import { appReducers } from '../../store';
import { Home } from './home';
import thunk from 'redux-thunk'
import { I18nextProvider } from 'react-i18next';
import i18next from 'i18next';
import common_en from "../../translations/en/common.json";

const store = createStore(
    appReducers,
    applyMiddleware(thunk)
)

i18next.init({
    interpolation: { escapeValue: false },
    lng: 'en',
    resources: {
        en: {
            common: common_en
        }
    },
})

it('renders without crashing', () => {
  const home = renderer.create(
      <I18nextProvider i18n={i18next}>
          <Provider store={store}>
            <Home />
          </Provider>
      </I18nextProvider>
  ).toJSON();

  expect(home).toMatchSnapshot();
});
