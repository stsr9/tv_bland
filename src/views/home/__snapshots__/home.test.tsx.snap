// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="splash-screen"
>
  <h1>
    TV BLAND
  </h1>
</div>
`;
